DiceKitTests/MultiplicationExpression_Tests.swift

```diff
@@ -0,0 +1,120 @@
+//
+//  MultiplicationExpression_Tests.swift
+//  DiceKit
+//
+//  Created by Brentley Jones on 7/18/15.
+//  Copyright © 2015 Brentley Jones. All rights reserved.
+//
+
+import XCTest
+import Nimble
+import SwiftCheck
+
+import DiceKit
+
+/// Tests the `MultiplicationExpression` type
+class MultiplicationExpression_Tests: XCTestCase {
+    
+}
+
+// MARK: - Equatable
+extension MultiplicationExpression_Tests {
+    
+    func test_shouldBeReflexive() {
+        property["reflexive"] = forAll {
+            (a: Int, b: Int) in
+            
+            let x = MultiplicationExpression(a, b)
+            
+            return x == x
+        }
+    }
+    
+    func test_shouldBeSymmetric() {
+        property["symmetric"] = forAll {
+            (a: Int, b: Int) in
+            
+            let x = MultiplicationExpression(a, b)
+            let y = MultiplicationExpression(a, b)
+            
+            return x == y && y == x
+        }
+    }
+    
+    func test_shouldBeTransitive() {
+        property["transitive"] = forAll {
+            (a: Int, b: Int) in
+            
+            let x = MultiplicationExpression(a, b)
+            let y = MultiplicationExpression(a, b)
+            let z = MultiplicationExpression(a, b)
+            
+            return x == y && y == z && x == z
+        }
+    }
+    
+    func test_shouldBeAbleToNotEquate() {
+        property["non-equal"] = forAll {
+            (a: Int, b: Int, c: Int, d: Int) in
+            
+            guard !(a == c && b == d) else { return true }
+            
+            let x = MultiplicationExpression(a, b)
+            let y = MultiplicationExpression(c, d)
+            
+            return x != y
+        }
+    }
+    
+    func test_shouldBeAnticommutative() {
```

Comment:
This is only written to this way because of things like dice generating a different value for each multiplicand result. If we can find a way to know if an expression (all the way down) is stable or not, we can allow this to be commutative in some cases.

cc: @yammada